cmd/rm-main.go

```diff
@@ -220,27 +241,35 @@ func removeRecursive(url string, isIncomplete bool, isFake bool, older int) erro
 		}
 
 		urlString := content.URL.Path
-		printMsg(rmMessage{
-			Key:  targetAlias + urlString,
-			Size: content.Size,
-		})
+		// print the rmMessage only if content is a bucket or object. clnt.List lists
+		// prefixes as well which unifies the S3 listing with folder listing - however
+		// these prefixes are not actual objects on s3 and the content.Time defaults to
+		// zero. Filter these out to prevent erroneous rmMessage being printed to console.
+		if !content.Time.IsZero() {
+			printMsg(rmMessage{
+				Key:  targetAlias + urlString,
+				Size: content.Size,
+			})
+		}
 
 		if !isFake {
 			sent := false
 			for !sent {
 				select {
 				case contentCh <- content:
 					sent = true
-				case pErr := <-errorCh:
+				case pErr, ok := <-errorCh:
+					if !ok {
+						close(contentCh)
+						return exitStatus(globalErrorExitStatus)
+					}
```

Comment:
We would need to exit on first error here.  This check leads to continuation.  Below is the right logic
```go
case pErr, ok := <-errorCh:
	if ok {
		errorIf(pErr.Trace(urlString), "Failed to remove `"+urlString+"`.")
		switch pErr.ToGoError().(type) {
		case PathInsufficientPermission:
			// Ignore Permission error.
			continue
		}
	}
	close(contentCh)
	return exitStatus(globalErrorExitStatus)
```

IMO, having `ok` is not required because `pErr` will be `nil` on closed channel and we exit anyway.